scripts/sudo.ps1

```diff
@@ -1,87 +1,78 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
+[CmdletBinding(DefaultParameterSetName = "Script")]
+param(
+	[Parameter(Mandatory, Position = 0, ParameterSetName = "Script")]
+	[scriptblock]$ScriptBlock,
 
-# open question - Should -NoProfile be used when invoking PowerShell
-BEGIN {
+	[switch]$NoProfile,
+
+	[Parameter(Mandatory, Position = 0, ParameterSetName = "Command")]
+	[string]$Command,
+
+	[Parameter(Position = 1, ValueFromRemainingArguments)]
+	[Alias("Args")]
+	[PSObject[]]$ArgumentList
+)
+begin {
 	if ($__SUDO_TEST -ne $true) {
-		$SUDOEXE = "sudo.exe"
-	}
-	else {
-		if ($null -eq $SUDOEXE) {
-			throw "variable SUDOEXE has not been set for testing"
-		}
+		$Env:SUDOEXE = "sudo.exe"
+	} elseif (!$Env:SUDOEXE) {
+			throw "Environment variable SUDOEXE has not been set for testing"
 	}
 
-	if ([Environment]::OSVersion.Platform -ne "Win32NT") {
+	if ($IsLinux -or $IsMacOS) {
```

Comment:
`$IsLinux` and `$IsMacOS` are only available since Powershell 7 / Core 6. The previous approach was more portable.

Reply:
The only place where they are _not_ available ... is where _neither_ is true, so this logic works without involving classes or string comparisons.

In PowerShell, when the variable doesn't exist, it's $null and thus, $false -- and as a bonus, this can be _faked_ for testing.